accepted/0000-pure-mode.md

```diff
@@ -0,0 +1,344 @@
+          │       │
+          │       └───> B @ * (peer dependency)
+          │
+          └───> B @ 2.0.0
+
+
+```
+
+#### Installation on disk
+
+```
+  root /
+   ┬
+   │
+   ├─> node_modules / .npm / ─┬─> A@1.0.0+B@1.0.0-21f95f7 / node_modules / A / ─┬─> [ content of package A ]
+   │                          │                                                 │
+   │                          │                                                 └─> node_modules / B ( symlink to ../../../../B@1.0.0-0d98ab/node_modules/B )
+   │                          │
+   │                          ├─> A@1.0.0+B@2.0.0-66fe689 / node_modules / A / ─┬─> [ content of package A ]
+   │                          │                                                 │
+   │                          │                                                 └─> node_modules / B ( symlink to ../../../../B@2.0.0-a2ea56/node_modules/B )
+   │                          │
+   │                          ├─> B@1.0.0-0d98ab / node_modules / B / ───> [ content of package B (v1) ]
+   │                          │
+   │                          └─> B@2.0.0-a2ea56 / node_modules / B / ───> [ content of package B (v2) ]
+   │
+   └─> workspaces / ─┬─> foo / ─┬─> [ content of workspace foo ]
+                     │          │
+                     │          └───> node_modules / ─┬─> A ( symlink to ../../../node_modules/.npm/A@1.0.0+B@1.0.0-21f95f7/node_modules/A )
+                     │                                │
+                     │                                └─> B ( synlink to ../../../node_modules/.npm/B@1.0.0-0d98ab/node_modules/B )
+                     │
+                     └─> bar / ─┬─> [ content of workspace bar ]
+                                │
+                                └───> node_modules / ─┬─> A ( symlink to ../../../node_modules/.npm/A@1.0.0+B@2.0.0-66fe689/node_modules/A )
+                                                      │
+                                                      └─> B ( synlink to ../../../node_modules/.npm/B@2.0.0-a2ia56/node_modules/B )
+
+```
+
+## Performance
+
+Compared to the current npm installation strategy, this proposal reduces package duplication, making the installation process faster. On a prototype, this installation strategy brought down the install time from 6 to 2 minutes on a large monorepo of 500+ workspaces.
+
+## Configuration
+
+The `pure-mode` is a project level setting, it does not make much sense to install some dependencies in `pure-mode` and others in `hoisted-mode`. For this reason, to configure a project in `pure-mode`, a setting has to be added to the `.npmrc` of the project and the lockfile has to be re-generated. A tool may be made available to convert `hoisted-mode` lockfiles to `pure-mode` lockfiles.
+
+## Compatibility
+
+### Different versions of npm
+
+As workspaces are supported only from npm version 7, there is no concern about making `pure-mode` compatible with versions olders than 7.0.0. 
```

Comment:
perhaps worth only supporting `package-lock` v3 or even a new v4 and above, then?

I honestly would not mind a feature this impactful being available starting in a new semver major to both make it clear which versions are supported by the feature and provide a good incentive to update.

Reply:
The new mode does bring very significant performance improvements,  however as an opt-in `pure-mode` can be introduced as a semver minor.

Reply:
The package-lock does not appear to need an upgrade to support `pure-mode` because it already support describing symlinks.

Shipping this mode in a major release may simplify the compatibility story, I will let the npm CLI team judge what is best here.

Reply:
We have decided to that using pure-mode will not affect the lock file at all, this mode will only affect how packages are laid out on disk but the lock file will stay the same.